### PR TITLE
LIB-16359 - Resolve issues with refreshing solr after move operations

### PR DESCRIPTION
**Pull request by bbpennel**
https://jira.lib.unc.edu/browse/LIB-16359
* Add status updating filter to move operation indexing, since inherited statuses were not clearing when moving out of a folder with permissions restrictions
* Invalidate path cache when an object is modified, to prevent stale value from preventing the path from updating in solr from a move operation if the same object had recently been indexed

### CODE REVIEW

File: services-camel-app/src/test/java/edu/unc/lib/boxc/services/camel/util/CacheInvalidatingProcessorTest.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import static org.mockito.Mockito.when;
 import static org.mockito.MockitoAnnotations.initMocks;
 
+import edu.unc.lib.boxc.model.api.services.ContentPathFactory;
```
Comment (lfarrell) [inline services-camel-app/src/test/java/edu/unc/lib/boxc/services/camel/util/CacheInvalidatingProcessorTest.java:26]: Should this be with the other edu imports?

File: services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/util/CacheInvalidatingProcessor.java [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 import static org.slf4j.LoggerFactory.getLogger;
 
+import edu.unc.lib.boxc.model.api.services.ContentPathFactory;
```
Comment (lfarrell) [inline services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/util/CacheInvalidatingProcessor.java:20]: Should this be with the other edu imports?

**Review by lfarrell — CHANGES_REQUESTED** [review]
Should you have to explicitly re-index after moving an object in/out of a restricted folder to see the inherited permissions change?

File: services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/util/CacheInvalidatingProcessor.java (hunk 2 above)
Comment (bbpennel) [reply services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/util/CacheInvalidatingProcessor.java:23]: I'm not sure why intellij put this here, guessing it didn't agree with eclipse about ordering. I've told it to organize imports

**Comment by bbpennel** [comment]
No, this PR is supposed to fix that particular bug. I've deployed the branch to test and am seeing restricted and inherited flags cleared on the moved object. What steps did you take to produce the issue? Did you deploy the services-camel application?

**Comment by lfarrell** [comment]
I hadn't deployed all the changes

**Review by lfarrell — APPROVED** [review]

**Pull request merged**